kernel/src/path.rs

```diff
@@ -97,22 +98,26 @@ impl AsUrl for Url {
     }
 }
 
+fn path_contains_delta_log_dir(mut path_segments: std::str::Split<'_, char>) -> bool {
+    path_segments.any(|p| p == "_delta_log")
+}
+
 impl<Location: AsUrl> ParsedLogPath<Location> {
     // NOTE: We can't actually impl TryFrom because Option<T> is a foreign struct even if T is local.
     #[internal_api]
     pub(crate) fn try_from(location: Location) -> DeltaResult<Option<ParsedLogPath<Location>>> {
         let url = location.as_url();
```

Comment:
It seems like we never return `Err`? Why keep this as `DeltaResult<Option<>>`

And in fact, why make this Return None for some things and LogFilePath::Unknown for others? The semantics of this function aren't clear to me. Could you either:
1) Make it always return ParsedLogPath. Weird files are considered unknown.
2) Document clearly what Ok(Some(Unknown)) vs Ok(None) means.

Reply:
Yeah, as I noted in the PR description, this isn't ideal. It should just return an option, but that's a much bigger change because it propagates up the stack and some of our iterator code is a bit hairy in how it deals with the return value here.

So I was proposing that we keep it like this in the short term where there's no semantic difference between `None` and `Some(Unknown)`, and then unify all that post 0.17. 

I can add a comment to that effect in the code if you think it's needed.